mirrord-layer/src/file/hooks.rs

```diff
@@ -34,14 +35,16 @@ pub(super) unsafe extern "C" fn open_detour(raw_path: *const c_char, open_flags:
         libc::open(raw_path, open_flags)
     } else {
         let open_options = OpenOptionsInternalExt::from_flags(open_flags);
-        let open_result = open(path, open_options);
+        let open_result = open(path.clone(), open_options);
 
-        open_result
-            .map_err(|fail| {
-                error!("Failed opening file with {:#?}", fail);
+        match open_result {
```

Comment:
Instead of `match`, you could do `map_err(From::from)`, something like this:

```rust
let (Ok(result) | Err(result)) = open.map_err(From::from);

result
```

And for functions where the safe wrapper doesn't return some number type, you could do `operantion.map(|something| .... )` before `map_err` to convert the result into the appropriate return value.